### PR TITLE
Form

### PR DESCRIPTION
**Pull request by anniviz**
Acceptance criteria

- Save data to local storage
- when there is no content for the main page, this is the first screen
- "zurück" button brings you back to main page and empties the form
- "erstellen" button is disabled until there is content for the main page
- "Hinzufügen" (start page) is fixed at the top of the screen (and not pushed down by cards)
- The url does not change

### CODE REVIEW

File: src/pages/FormPage.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        Medikamente:
+        <Textarea
+          name="meds"
+          rows="15"
+          placeholder={placeholderText}
+          onChange={handleChange}
+        />
+      </Label>
+      <Grid>
+        <Button onClick={() => onClick('medication')} isActive type="button">
```
Comment (Morten-Moeller) [inline src/pages/FormPage.js]: could the onClick not be called onNavigate?

File: src/pages/FormPage.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    event.preventDefault()
+    if (isDisabled) return
+    const form = event.target
+    const time = form.elements.time.value
+    if (!validateTime(time)) {
+      setIsTimeValid(false)
+      form.elements.time.focus()
+      return
+    }
+    const meds = form.elements.meds.value
```
Comment (Morten-Moeller) [inline src/pages/FormPage.js]: you could de-structure form.elements  in the beginning

File: src/pages/FormPage.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    const medsArrayWithId = meds
+      .split('\n')
+      .map(medName => ({ id: uuidv4(), medName: medName }))
+
+    onSubmit({ id: uuidv4(), time: time, meds: medsArrayWithId })
```
Comment (Morten-Moeller) [inline src/pages/FormPage.js]: if you have the key and the value the same, you can just use `time`

File: src/pages/FormPage.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    setActivePage('medication')
+  }
+
+  function handleChange(event) {
+    const { name, value } = event.target
+    setMedGroupInputs({ ...medGroupInputs, [name]: value })
+  }
+
+  function validateTime(time) {
+    const timeFormat = /^\d{1,2}:\d{2}$/
```
Comment (Morten-Moeller) [inline src/pages/FormPage.js]: 99:99 is a valid time with this expression. Maybe you wont to go for:
`^([0-1][0-9]|2[0-3]):([0-5][0-9])$`

File: src/pages/MedicationPage.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  )
+
+  function convertToMinutes(time) {
+    const timeArray = time.split(':')
+    const minutes = timeArray[0] * 60 + timeArray[1]
+    console.log(time + ': ' + minutes)
+    return Number(minutes)
+  }
+
+  function formatDate(date) {
```
Comment (Morten-Moeller) [inline src/pages/MedicationPage.js:48]: Maybe you want to have a look on [toLocaleDateString](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString)

In my opinion that should be way shorter.

**Review by Morten-Moeller — COMMENTED** [review]
Well done, keep on rolling 👾
Only a few small comments 🤓

File: src/pages/FormPage.js (hunk 2 above)
Comment (anniviz) [reply src/pages/FormPage.js]: I like

File: src/pages/FormPage.js (hunk 4 above)
Comment (anniviz) [reply src/pages/FormPage.js]: oops, yes!

File: src/pages/FormPage.js (hunk 2 above)
Comment (anniviz) [reply src/pages/FormPage.js]: actually, I think we need to talk about this, tomorrow 😉

File: src/pages/FormPage.js (hunk 4 above)
Comment (anniviz) [reply src/pages/FormPage.js]: you missed a case and it needs to be like this:
`^([0-9]|[0-1][0-9]|2[0-3]):([0-5][0-9])$`

**Pull request merged**